File: mobile/lib/data/models/entrance.dart
class Entrance {
  const Entrance({
    required this.id,
    required this.buildingId,
    required this.latitude,
    required this.longitude,
    required this.floor,
    required this.name,
  });

  final String id;
  final String buildingId;
  final double latitude;
  final double longitude;
  final int floor;
  final String name;

  factory Entrance.fromJson(Map<String, dynamic> json) => Entrance(
        id: json['id'] as String,
        buildingId: json['buildingId'] as String,
        latitude: (json['latitude'] as num).toDouble(),
        longitude: (json['longitude'] as num).toDouble(),
        floor: (json['floor'] as num?)?.toInt() ?? 1,
        name: json['name'] as String? ?? 'Main Entrance',
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'buildingId': buildingId,
        'latitude': latitude,
        'longitude': longitude,
        'floor': floor,
        'name': name,
      };
}

File: backend/app/models/entrance.py
from sqlalchemy import Float, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.database.session import Base


class Entrance(Base):
    __tablename__ = "entrances"

    id: Mapped[str] = mapped_column(String, primary_key=True)
    building_id: Mapped[str] = mapped_column(String, index=True, nullable=False)
    latitude: Mapped[float] = mapped_column(Float, nullable=False)
    longitude: Mapped[float] = mapped_column(Float, nullable=False)
    floor: Mapped[int] = mapped_column(Integer, default=1)
    name: Mapped[str] = mapped_column(String, default="Main Entrance")
